api/hardwaremanagement/v1alpha1/conditions.go

```diff
@@ -23,3 +23,22 @@ const (
 	// fulfilled.
 	FailedCondition = "Failed"
 )
+
+type ConditionType string
+
+// The following constants define the different types of conditions that will be set
+const (
+	Provisioned ConditionType = "Provisioned"
+	Unknown     ConditionType = "Unknown" // indicates the condition has not been evaluated
+)
+
+type ConditionReason string
+
+// The following constants define the different reasons that conditions will be set for
+const (
+	InProgress     ConditionReason = "InProgress"
+	Completed      ConditionReason = "Completed"
+	Unprovisioned  ConditionReason = "Unprovisioned"
+	Failed         ConditionReason = "Failed"
+	NotInitialized ConditionReason = "NotInitialized"
+)
```

Comment:
These would be more convenient if setup similar to the condition types and reasons as in:
https://github.com/openshift-kni/oran-o2ims/blob/main/internal/controllers/utils/conditions.go#L8-L48

Something like:
```
// ConditionReason is a string representing the condition's reason
type ConditionReason string

// The following constants define the different reasons that conditions will be set for the resource
var ResourceConditionReasons = struct {
	Completed  ConditionReason
	InProgress ConditionReason
	Failed     ConditionReason
}{
	Completed:  "Completed",
	InProgress: "InProgress",
	Failed:     "Failed",
}
```

Rather than a random collection of constants, they're grouped and referenced like:
```
hwmgmtv1alpha1.ResourceConditionReasons.InProgress
```